.devcontainer.json

```diff
@@ -0,0 +1,16 @@
+{
```

Comment:
@mumichae should we include this file if we're including the Dockerfile?

Reply:
This seems to be vscode specific. @Moomboh Could you elaborate on how os-specific this setup is?

Reply:
It's independent of the Dockerfile. It is also not vscode specific per-se although VS code is the only IDE currently supporting it fully, see https://containers.dev/supporting. There also exists a barebones CLI tool though: https://github.com/devcontainers/cli.
It is not in any way OS specific, only depends on Docker and VS code. You might run into docker volume mount file permission issues on Windows file system (without WSL2), but that's a general Windows+Docker issue and has nothing to do with the setup.
I would recommend keeping it in the repo, it won't hurt anyone not working with devcontainers and makes things a lot easier for people who do, but it's totally up to you.
On the other hand, if you don't want to include it the Dockerfile might need refactoring into [multistage build](https://docs.docker.com/build/building/multi-stage/) and split it into `base` (minimum for running sparsecca) `test` and `dev` stages respectively.

Reply:
Alright, let's leave it!